src/v/raft/consensus.h

```diff
@@ -161,7 +161,9 @@ class consensus {
     model::term_id term() const { return _term; }
     group_configuration config() const;
     const model::ntp& ntp() const { return _log.config().ntp(); }
-    clock_type::time_point last_heartbeat() const { return _hbeat; };
+    clock_type::time_point last_get_append_entries_req_timestamp() const {
+        return _hbeat;
```

Comment:
Hmm, there is a case where consensus does this:
```
                _hbeat = clock_type::time_point::min();
```

So _hbeat isn't exactly when we got our last heartbeat, it has other behaviors.

I haven't read the following commits yet: this might not matter if the consuming code is aware of the behavior, but it if it does a naive "time since `last_get_append_entries_req_timestamp` then that could be a problem.

Reply:
@mmaslankaprv I am not sure about this place and impact for check

Reply:
As I understand it is signal that consensus just started. So I think i this case we can return that node is not isolated. And wait next round of communication?

Reply:
Move this check to just check leader